File: Euclid/Messages/Incoming/PLACEITEMFROMSTRIP.cs
using Euclid.Game;
using Euclid.Network.Streams;
using Euclid.Util.Extensions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Euclid.Messages.Incoming
{
    class PLACEITEMFROMSTRIP : IMessageEvent
    {
        public void Handle(Player player, Request request)
        {
            if (player.RoomEntity.Room == null)
                return;

            Room room = player.RoomEntity.Room;
            var placementData = request.Content.Split(' ');

            if (!placementData[0].IsNumeric())
                return;

            int itemId = int.Parse(placementData[0]);
            Item item = player.Inventory.GetItem(itemId);

            if (item == null)
                return;

            if (room == null || !room.HasRights(player.Details.Id))
            {
                return;
            }


            string[] wallPositionData = request.Content.Split("/")[0].Split(" ");
            string wallPosition = $"{wallPositionData[1]} {wallPositionData[2]}";

            room.FurnitureManager.AddItem(item, wallPosition: wallPosition, player: player);

            player.Inventory.RemoveItem(item);
        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/PLACEITEMFROMSTRIP.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.game.item.Item;
import euclid.game.room.Room;
import euclid.messages.IMessageEvent;
import euclid.network.streams.Request;
import euclid.util.extensions.StringExtensions;

public class PLACEITEMFROMSTRIP implements IMessageEvent {

    @Override
    public void handle(Player player, Request request) {
        if (player.getRoomEntity().getRoom() == null) {
            return;
        }

        Room room = player.getRoomEntity().getRoom();
        String[] placementData = request.getContent().split(" ");

        if (placementData.length < 1 || !StringExtensions.isNumeric(placementData[0])) {
            return;
        }

        int itemId = Integer.parseInt(placementData[0]);
        Item item = player.getInventory().getItem(itemId);

        if (item == null) {
            return;
        }

        if (room == null || !room.hasRights(player.getDetails().getId())) {
            return;
        }

        String[] wallPositionData = request.getContent().split("/")[0].split(" ");
        String wallPosition = wallPositionData[1] + " " + wallPositionData[2];

        room.getFurnitureManager().addItem(item, null, wallPosition, player);

        player.getInventory().removeItem(item);
    }
}
